From bf9b002053b7f8def1474027cbd7b9efda4f237b Mon Sep 17 00:00:00 2001
From: Lin Liu <Lin.Liu01@cloud.com>
Date: Wed, 26 Mar 2025 06:13:58 +0000
Subject: [PATCH 01/10] CP-54081: Remove interface-rename-sideway service

With netdev ordering work delegated to networkd, host installer
no longer need to rename netdev, thus, no longer requires
interface-rename-sideway service

Signed-off-by: Lin Liu <Lin.Liu01@cloud.com>
---
 Makefile                                 |  2 --
 startup/interface-rename-sideway         | 22 ----------------------
 startup/interface-rename-sideway.service | 10 ----------
 3 files changed, 34 deletions(-)
 delete mode 100644 startup/interface-rename-sideway
 delete mode 100644 startup/interface-rename-sideway.service

diff --git a/Makefile b/Makefile
index d507958..303aca9 100644
--- a/Makefile
+++ b/Makefile
@@ -63,7 +63,6 @@ install:
 	    $(DESTDIR)/etc/dracut.conf.d \
 	    $(DESTDIR)/etc/systemd/system/systemd-udevd.d
 
-	$(INSTALL) -m755 startup/interface-rename-sideway $(DESTDIR)/$(INSTALLER_DIR)
 	$(INSTALL) -m644 startup/functions $(DESTDIR)/$(INSTALLER_DIR)
 	$(INSTALL) -m644 startup/bnx2x.conf $(DESTDIR)/etc/modprobe.d/
 	$(INSTALL) -m644 startup/blacklist $(DESTDIR)/etc/modprobe.d/installer-blacklist.conf
@@ -71,7 +70,6 @@ install:
 	$(INSTALL) -m644 startup/iscsi-modules $(DESTDIR)/etc/modules-load.d/iscsi.conf
 	$(INSTALL) -m755 startup/preinit startup/S05ramdisk $(DESTDIR)/$(INSTALLER_DIR)/
 	$(INSTALL) -m644 startup/systemd-udevd_depmod.conf $(DESTDIR)/etc/systemd/system/systemd-udevd.d/installer.conf
-	$(INSTALL) -m644 startup/interface-rename-sideway.service $(DESTDIR)/$(SERVICE_DIR)
 
  # Generate a multipath configuration from the installed copy, removing
  # the blacklist and blacklist_exception sections.
diff --git a/startup/interface-rename-sideway b/startup/interface-rename-sideway
deleted file mode 100644
index 490f576..0000000
--- a/startup/interface-rename-sideway
+++ /dev/null
@@ -1,22 +0,0 @@
-#!/bin/sh
-#
-# interface rename initialization
-#
-
-udevadm settle timeout=180
-udevadm trigger
-udevadm settle timeout=180
-
-cd /sys/class/net
-for i in eth*; do
-  if ! ip link show $i | grep -q '[,<]UP[,>]'; then
-      # len(name) cannot be greater than 15. Using a 4 digit random number
-      # allows for 100 (eth0-eth99) devices to have a temp name without going
-      # over 15 byte limit
-      name="side-$((RANDOM%9999))-$i"
-      while [ -e "/sys/class/net/$name" ]; do
-          name="side-$((RANDOM%9999))-$i"
-      done
-      ip link set $i name "$name"
-  fi
-done
diff --git a/startup/interface-rename-sideway.service b/startup/interface-rename-sideway.service
deleted file mode 100644
index 05c88a0..0000000
--- a/startup/interface-rename-sideway.service
+++ /dev/null
@@ -1,10 +0,0 @@
-[Unit]
-Description=Prepare for interface rename
-Before=multi-user.target
-Before=graphical.target
-Before=network-pre.target
-
-[Service]
-ExecStart=/opt/xensource/installer/interface-rename-sideway
-RemainAfterExit=yes
-Type=oneshot
-- 
2.39.5

